addon/components/nrg-dropdown.js

```diff
@@ -299,112 +220,164 @@ export default Component.extend(Validation, EKMixin, EKFirstResponderOnFocusMixi
       return false;
     }
     return s1.toLowerCase().indexOf(s2.toLowerCase()) != -1;
-  },
+  }
 
-  mouseDown(evt) {
-    const isMultiSelection = evt.target.closest('[data-dropdown-multi-selection]');
-    const isDropdownItem = evt.target.closest('[data-dropdown-item]');
-    if (isMultiSelection || isDropdownItem) {
-      return false;
-    }
-    const isDropdownIcon = evt.target.closest('.dropdown.icon');
-    const clickedInsideDropdownElement = evt.target.closest('.dropdown');
-    if (this.search && !this.isOpen) {
-      this.openDropdown();
-    } else if (!this.search || isDropdownIcon) {
-      if (this.isOpen && (isDropdownIcon || !clickedInsideDropdownElement)) {
-        this.closeDropdown();
-      } else {
-        this.openDropdown();
-      }
-    }
-  },
+  onSelect(option) {
+    this._onChange(option);
+  }
 
-  focusIn() {
-    if (this._dropdownDisabled) {
-      return;
-    }
-    if (this.showOnFocus) {
-      this.openDropdown()
-    }
-    if (this.search) {
-      this.focusInput(true);
+  isCurrentlySelected(option) {
+    if (this.args.multiple) {
+      if (!Array.isArray(this.value)) {
+        return false;
+      }
+      return this.value.indexOf(option) != -1;
+    } else {
+      return option === this.value;
     }
```

Comment:
Unnecessary `else` block
```suggestion
    }
    return option === this.value;
```